tests/jupyterhub_api/test_check_api.py

```diff
@@ -0,0 +1,77 @@
+
+
+import json
+import requests
+import subprocess
+import os
+import docker
+import sys
+sys.path.append("..")
+import time
+from logger import Logger, LOG_FOLDER, LOG_EXTENSION
+from SessionUtils import Test, JupyterhubTest
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+import argparse
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Arguments', formatter_class = argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--port", dest="port", type=int,
+                        required = True,
+                        )
+    parser.add_argument("--base_path", dest="base",
+                        required=False,
+                        help='base path')
+
+    args = parser.parse_args()
+    return args
+
+
+class CheckAPI(JupyterhubTest):
+    """
+    Test if a api is reachable
+
+    """
+    def __init__(self, port, base_path, verify):
+        JupyterhubTest.__init__(self, port, base_path,  verify)
+        self.ref_test_name = "APIReachable"
+        super().log_params()
+        self.exit = 0
```

Comment:
Why do you set the exit code as a value of the class?
You should get the value by calling `exit_code`, but this runs the code that checks the real value, so I don't see the reason why you set it beforehand.

Reply:
To avoid getting null values while calling exit code function, I removed it from there.

Reply:
How would you get null values? You call `exist_code` and you receive an output. If it's null or something, you return 0 instead... You just don't need to set it in the object.